src/super_gradients/training/processing/processing.py

```diff
@@ -84,20 +87,44 @@ def infer_image_input_shape(self) -> Optional[Tuple[int, int]]:
         """
         return None
 
+    @property
+    @abstractmethod
+    def resizes_image(self) -> bool:
+        """Return True if the processing resizes the image, False otherwise."""
+        pass
+
 
 @register_processing(Processings.ComposeProcessing)
 class ComposeProcessing(Processing):
     """Compose a list of Processing objects into a single Processing object."""
 
-    def __init__(self, processings: List[Processing]):
+    def __init__(self, processings: List[Processing], skip_image_resizing: bool = False):
+        """
+        :param processings:     List of Processing objects to compose.
+        :param skip_image_resizing: If True, skip the image resizing processings.
+        """
         self.processings = processings
+        self.include_image_resizing = not skip_image_resizing
+
+    def disable_image_resizing(self):
```

Comment:
I don't like the idea of modifying processing params in-place. I think it is error-prone and could be done differently in a (subjectively) simple manner:

We have list of processing operations. We know the subset of operations that should not be run if skip resizing is enabled: LongestMaxSize + Pad.

So to generate a new processing transforms we should iterate over existing transforms, look for these two transfom classes and exclude then, adding a KeypointsAutoPadding instead.

This operation can happen in get_pipeline method so that logic of the Pipeline class would be unchanged.

Reply:
Introducing a `get_equivalent...` method  [6cf8cc9](https://github.com/Deci-AI/super-gradients/pull/1605/commits/6cf8cc9e3446e8b27179fb75e4aa550569a4d182)
And moving the skip logic to `_get_pipeline` [2922599](https://github.com/Deci-AI/super-gradients/pull/1605/commits/292259960ecca72a0d5afdc46149ce624e666d24)